ci-operator/step-registry/workers-rhel/ibmcloud-provision/workers-rhel-ibmcloud-provision-commands.sh

```diff
@@ -0,0 +1,125 @@
+fi
+echo "Reading variables from ${VPC_CONFIG}..."
+vpcName=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.vpcName')
+resource_group=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.resourceGroupName')
+infra_id=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+echo "Using region: ${region}  resource_group: ${resource_group} vpc: ${vpcName}"
+
+ibmcloud_login
+
+${IBMCLOUD_CLI} target -g ${resource_group} -r ${region}
+
+workdir=`mktemp -d`
+
+# Get bastion RG info
+vnet_RG=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.networkResourceGroupName')
+echo "vpc RG: ${vnet_RG}"
+
+#when run more jobs at the same time, fail to create key with the same fingerprint, so create the key (ci-qe-key) in Default group for all test. 
+${IBMCLOUD_CLI} is keys --all-resource-groups
+
+preDefaultKey="ci-qe-key"
+
+#add one more sg rule for the ansible test 
+sg=${infra_id}-sg-cluster-wide
+# ${IBMCLOUD_CLI} is security-group-rule-add $sg inbound tcp --remote "0.0.0.0/0" --port-min=22623 --port-max=22623
+sg1=${infra_id}-sg-openshift-net
+
+# Start to provision rhel instances from template in existing VPC and NSG
+for count in $(seq 1 ${RHEL_WORKER_COUNT}); do
+  echo "$(date -u --rfc-3339=seconds) - Provision ${infra_id}-rhel-${count} ..."
+  # Get computeSubnet
+  idx=$((count-1))
+  subnet=$(ibmcloud is vpc ${vpcName} --show-attached --output JSON | jq -c -r --argjson n $idx '[.subnets[] | select(.name|test("compute")) | .name][$n]')
+  zone=$(ibmcloud is subnet ${subnet} --output JSON | jq -r '.zone.name')
+  volume="${infra_id}-vol-$count"
+  vmName=${infra_id}-rhel-${count}
+  echo "computeSubnet ${subnet} zone ${zone} volume $volume vmName $vmName"
+
+  volumeJson=$(jq -n \
+    --arg vn "$volume" \
+    --argjson size ${RHEL_VM_DISK_SIZE} \
+    '{"name": $vn, "volume": {"capacity": $size, "profile": {"name": "general-purpose"}}}')
+  cmd="${IBMCLOUD_CLI} is instance-create $vmName ${vpcName} ${zone} ${RHEL_VM_SIZE} ${subnet} --image ${RHEL_IMAGE} --keys $preDefaultKey --sgs ${sg},${sg1} --boot-volume '${volumeJson}'"
+
+  echo "Creating RHEL VM: ${cmd}"
+  eval "${cmd}"
+  sleep 60
+
+  insFile="${workdir}/${vmName}.json"
+  echo "create instance ${vmName}, recored in ${insFile} ..."
+  ${IBMCLOUD_CLI} is instance ${vmName} --output JSON > ${insFile}
+  cat ${insFile}
+  echo "INFO" "Created instance ${vmName} status: $(jq -r '.status' ${insFile})"
+  rhel_node_ip="$(jq -r '.network_interfaces[0].primary_ip.address' ${insFile})"
+  echo "Ip address is ${rhel_node_ip}"
+  
+  if [ "x${rhel_node_ip}" == "x" ]; then
+    echo "Unable to get ip of rhel instance ${infra_id}-rhel-${count}!"
+    exit 1
+  fi
+   
+  echo ${rhel_node_ip} >> /tmp/rhel_nodes_ip
+done
+
+cp /tmp/rhel_nodes_ip "${ARTIFACT_DIR}"
+
+#Get bastion info
+BASTION_SSH_USER=$(cat "${SHARED_DIR}/bastion_ssh_user")
+bastion_info_yaml="${SHARED_DIR}/bastion-info.yaml"
+BASTION_PUBLIC_IP=$(yq-go r "${bastion_info_yaml}" 'publicIpAddress')
```

Comment:
bastion public ip is already saved, why not:
```
BASTION_PUBLIC_IP=$(cat "${SHARED_DIR}/bastion_public_address")
```

Reply:
I will update it.